src/Morpho.sol

```diff
@@ -377,11 +377,16 @@ contract Morpho is IMorpho {
         uint256 badDebtShares;
         if (position[id][borrower].collateral == 0) {
             badDebtShares = position[id][borrower].borrowShares;
-            uint256 badDebt = badDebtShares.toAssetsUp(market[id].totalBorrowAssets, market[id].totalBorrowShares);
-            market[id].totalSupplyAssets -= badDebt.toUint128();
-            market[id].totalBorrowAssets -= badDebt.toUint128();
-            market[id].totalBorrowShares -= badDebtShares.toUint128();
             position[id][borrower].borrowShares = 0;
+
+            uint128 totalBorrowAssets = market[id].totalBorrowAssets;
+            uint128 badDebt = UtilsLib.min(
+                totalBorrowAssets, badDebtShares.toAssetsUp(totalBorrowAssets, market[id].totalBorrowShares)
+            ).toUint128();
```

Comment:
Here is a failing test for this change:
```solidity

    function testBadDebtDoesNotOverflow() public {
        {
            uint256 collateralAmount = 10 ether;
            uint256 loanAmount = 1 ether;
            _supply(loanAmount);

            collateralToken.setBalance(BORROWER, collateralAmount);
            vm.startPrank(BORROWER);
            morpho.supplyCollateral(marketParams, collateralAmount, BORROWER, hex"");
            morpho.borrow(marketParams, loanAmount, 0, BORROWER, BORROWER);
            // Trick to inflate shares, so that the computed bad debt is greater than the total debt of the market.
            morpho.borrow(marketParams, 0, 1, BORROWER, BORROWER);
            vm.stopPrank();

            oracle.setPrice(1e36 / 100);

            loanToken.setBalance(LIQUIDATOR, loanAmount);
            vm.prank(LIQUIDATOR);
            morpho.liquidate(marketParams, BORROWER, collateralAmount, 0, hex"");
        }
    }
```

Reply:
So what should we do?

Reply:
Oh I get it, we should not cast `badDebt` to `uint128` is that it?

Reply:
Btw casting it to `uint128` could be unchecked because `badDebt <= totalBorrowAssets <= type(uint128).max`

Reply:
> So what should we do?

Let's add this test !

Reply:
It's also valid for `badDebtShares` btw!